common/src/KokkosFFT_traits.hpp

```diff
@@ -107,6 +110,111 @@ template <typename ViewType>
 inline constexpr bool is_admissible_view_v =
     is_admissible_view<ViewType>::value;
 
+template <typename ExecutionSpace, typename ViewType, typename Enable = void>
+struct is_operatable_view : std::false_type {};
+
+template <typename ExecutionSpace, typename ViewType>
+struct is_operatable_view<
+    ExecutionSpace, ViewType,
+    std::enable_if_t<
+        Kokkos::is_execution_space_v<ExecutionSpace> &&
+        is_admissible_view_v<ViewType> &&
+        Kokkos::SpaceAccessibility<
+            ExecutionSpace, typename ViewType::memory_space>::accessible>>
+    : std::true_type {};
+
+/// \brief Helper to check if a View is an acceptable View for Kokkos-FFT and
+/// memory space is accessible from the ExecutionSpace
+template <typename ExecutionSpace, typename ViewType>
+inline constexpr bool is_operatable_view_v =
+    is_operatable_view<ExecutionSpace, ViewType>::value;
+
+// Traits for binary operations
+template <typename T1, typename T2, typename Enable = void>
+struct have_same_precision : std::false_type {};
+
+template <typename T1, typename T2>
+struct have_same_precision<
+    T1, T2,
+    std::enable_if_t<!Kokkos::is_view_v<T1> && !Kokkos::is_view_v<T2> &&
+                     std::is_same_v<base_floating_point_type<T1>,
+                                    base_floating_point_type<T2>>>>
+    : std::true_type {};
+
+template <typename InViewType, typename OutViewType>
+struct have_same_precision<
```

Comment:
Not sure if "precision" is the correct name. Is it the same base floating point type?

Reply:
Sure, I will change the name accordingly.